core/src/main/java/io/aiven/klaw/service/ClusterApiService.java

```diff
@@ -127,8 +125,11 @@ public ClusterApiService(ManageDatabase manageDatabase) {
     this.manageDatabase = manageDatabase;
   }
 
-  private RestTemplate getRestTemplate() {
-    if (clusterConnUrl.toLowerCase().startsWith("https")) {
+  private RestTemplate getRestTemplate(String clusterConnectivityUrl) {
+    if (StringUtils.isEmpty(clusterConnectivityUrl)) {
+      clusterConnectivityUrl = clusterConnUrl;
+    }
+    if (clusterConnectivityUrl.toLowerCase().startsWith("https")) {
       if (this.httpsRestTemplate == null) {
```

Comment:
during testing I noticed that if I test http and then go to test https it wont reload the restTemplate because it is not null.
Perhaps we need a check here that checks to see if the urls clusterConnectivityUrl and clusterConnUrl are different and if they are reload the restTemplate even if it is not null. That way during normal operations it wont reload but when testing/setting up the check it will?
if (this.httpsRestTemplate == null) {

Reply:
This was a separate issue with my local env.